tests/overload/t13549.nim

```diff
@@ -0,0 +1,16 @@
+discard """
+  output: '''b
+c'''
+"""
+
+proc foo[T](x: T) = echo "a"
+
+
+proc foo[T: tuple](x: T) = echo "b"
+proc foo(x: tuple) = echo "b"
```

Comment:
Please only add one of these to the testcase, since technically these *should* give an ambiguous call error.